src/utils/types.ts

```diff
@@ -235,7 +247,7 @@ export interface CalendarViewProps extends CommonProps {
   newSelectedRangeEnd: Date | undefined;
   selectedMultiDates: Record<string, Date | undefined>;
   highlightsMap: Record<string, 1>;
-  viewDate: string | undefined;
+  viewDate: Date | undefined;
```

Comment:
Changed to string so that during re-render the props compare does not return true if the date object is for the same date value.

Scenario: If `viewDate` is updated after press of a `shortcutButton` and the user changes the month, we need to pass a new `viewDate` object into the props when the `shortcutButton` is pressed again (since a `string` `viewDate` type value remains the same and comp will not re-render). Let me know if this is not the correct approach.

Reply:
Will have to look into this. The reason I changed it to string from Date was that if the user provides us something like this <Calendar viewDate={new Date()} /> Then for every render this will be a new object so it will trigger all the hooks that are dependent on it all the time. 

Right now we update viewDate based on a number of different options in different scenario.

```ts
  // This just tries to find a month to show based on a number
  // of factors for the initial render only
  const [monthInView, setMonthInView] = useState<MonthIndices>(
    () =>
      getDateToShow({
        isNormalView: isNormalView,
        isMultiSelectorView: isMultiSelectorView,
        isRangeSelectorView: isRangeSelectorView,
        selectedDate: selectedDate,
        selectedRangeStart: selectedRangeStart,
        selectedMultiDates: selectedMultiDates,
        viewDate: viewDate ? fromString(viewDate) : undefined,
        minAllowedDate: minAllowedDate ? fromString(minAllowedDate) : undefined,
        maxAllowedDate: maxAllowedDate ? fromString(maxAllowedDate) : undefined,
      }).getMonth() as MonthIndices,
  );

  // This just tries to find a year to show based on a number
  // of factors for the initial render only
  const [yearInView, setYearInView] = useState(
    getDateToShow({
      isNormalView: isNormalView,
      isMultiSelectorView: isMultiSelectorView,
      isRangeSelectorView: isRangeSelectorView,
      selectedDate: selectedDate,
      selectedRangeStart: selectedRangeStart,
      selectedMultiDates: selectedMultiDates,
      viewDate: viewDate ? fromString(viewDate) : undefined,
      minAllowedDate: minAllowedDate ? fromString(minAllowedDate) : undefined,
      maxAllowedDate: maxAllowedDate ? fromString(maxAllowedDate) : undefined,
    }).getFullYear(),
  );

  useLayoutEffect(() => {
    if (showDualCalendar && isSecondary) {
      const nextMonth = getNextMonth(monthInView);
      setMonthInView(nextMonth);
      setYearInView(nextMonth === 0 ? getNextYear(yearInView) : yearInView);
    }
    // we are intentionally missing monthInView and yearInView deps
    // because we only want to auto compute secondary month when the props were changed the first time
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [showDualCalendar, isSecondary]);

  // updating view because view prop change
  useEffect(() => {
    if (viewDate && isValid(fromString(viewDate))) {
      setMonthInView(fromString(viewDate).getMonth() as MonthIndices);
      setYearInView(fromString(viewDate).getFullYear());
    }
  }, [viewDate]);

  // updating view when selected date change
  useEffect(() => {
    if (isValid(selectedDate)) {
      setMonthInView(selectedDate.getMonth() as MonthIndices);
      setYearInView(selectedDate.getFullYear());
    }
  }, [selectedDate]);

  // updating view when first multi is selected, after that
  // we don't update
  useEffect(() => {
    const dates = Object.keys(selectedMultiDates)
      .map((str) => selectedMultiDates[str])
      .filter((d) => isValid(d));

    if (dates.length === 1 && dates[0]) {
      setMonthInView(dates[0].getMonth() as MonthIndices);
      setYearInView(dates[0].getFullYear());
    }
  }, [selectedMultiDates]);
  ```
  
When we get a viewDate for the first render this component gives full preference to it and ignore all other variables.

After that it sort of manages it internally based on user actions.

So if the user gives us the viewDate prop in this way <Calendar viewDate={new Date()} /> then this component will think that for every render it gets a fresh new view date so for every render it would give preference to viewDate which basically means the view date will not be managed internally which would result in weird behaviour. 

This why I changed it to string. So even if Calendar gets it as a new object every time, since we convert it to a string before giving it to CalendarView it will not consider it a new date since string comparison works differently than object comparison.

Reply:
Maybe keep  `viewDate` (string) then and have another property? If it's only supposed to have authority in the first render and then managed internally. 

The other property `explicitViewDate` (date) can be used by the user or the shortcut parent to explicitly give a view when required. This prop will not meddle with other states and this way the user will also have some control over which view they want to change to if they explicitly ask the calendar to do so.

Reply:
Updated the API to change the view date. See here - https://github.com/natscale/react-calendar/blob/63990590a7b4e7363d9d77b94bb890950462c50b/docs/src/App.tsx#L177-L186

No longer uses a prop.